ebs/ebs_service.go

```diff
@@ -424,10 +446,20 @@ func (s *ebsService) GetVolumeByName(volumeName, dcName string) (*ec2.Volume, er
 		return nil, parseAwsError(err)
```

Comment:
Can you check the way the error is returned ? The caller is expecting a util.ConvoyDriverErr
See https://github.com/medallia/convoy/blob/master/ebs/ebs.go#L690-L695
This can cause the error to be ignored.
This should use util.NewConvoyDriverErr
Might be worth checking everywhere to cleanup this possible issue.

Reply:
Just checked, parseAwsError() returns a util.NewConvoyDriverErr() which returns a util.ConvoyDriverErr{}, should be ok but i'll do a quick check of any type assertions on errors.

Reply:
Great, I should have followed the function :-)